### PR TITLE
Alerting: Fix Slack notification preview

### PR DESCRIPTION
**Pull request by ekrucio**
**What this PR does / why we need it**:
When we're mentioning users, groups or channels the notification body is missing "text" field, which is required by Slack's Notification API -> https://api.slack.com/reference/messaging/payload .

**Which issue(s) this PR fixes**:
Fixes -> https://github.com/grafana/grafana/issues/42768

Notification with correctly displayed text -> 
Screenshot 2022-06-05 at 20 39 47

This PR does not introduce regression on earlier versions. Tested on 9.0.0. Also tested with legacy alerting and with the new concept - Unified Alerting.

### CODE REVIEW

**Comment by joeblubaugh** [comment]
Thanks for making this submission! I've verified on 9.0 as well and I'll set the PR so it merges into the 9.0 release.

**Comment by ekrucio** [comment]
> Thanks for making this submission! I've verified on 9.0 as well and I'll set the PR so it merges into the 9.0 release.

@joeblubaugh Glad you're okay with it! Sorry but I'm not aware with your PR flow, so what're the next steps? Would we merge it soon, or I should check it frequently and rebase/fix conflicts when they arise until it's merged?

**Comment by joeblubaugh** [comment]
@ekrucio I’ll do a review and then merge it if everything looks good. I was waiting for CI to pass before reviewing the submission.

**Review by santihernandezc — CHANGES_REQUESTED** [review]
Thanks for taking the time to fix this, I tested it and it's working fine 🚀 
I requested some changes, and the CI pipeline is failing, so check if merging `grafana:main` into your branch solves that.

File: pkg/services/ngalert/notifier/channels/slack.go
Comment (ekrucio) [reply pkg/services/ngalert/notifier/channels/slack.go:345]: Absolutely agree, will fix it.

File: pkg/services/ngalert/notifier/channels/slack.go
Comment (ekrucio) [reply pkg/services/ngalert/notifier/channels/slack.go:168]: I agree, but my logic was that if no text is provided there would be an empty string, which the Slack API would accept and should seng a notification with only a title and body, but if we omit it, then we're back to the beggining, with the _"content could not be displayed"_ message.

**Review by joeblubaugh — DISMISSED** [review]
This "request changes" is to block merges until we get to a safe point in the Grafana release process - it doesn't have anything to do with the contents of the PR, but we need to get a new branch set up so we don't accidentally add this to 9.0.0 in the middle of the final beta.

File: pkg/services/ngalert/notifier/channels/slack.go
Comment (santihernandezc) [inline pkg/services/ngalert/notifier/channels/slack.go:306]: Could you do the same for the `Text` field inside `Attachments`?

**Comment by grobinson-grafana** [comment]
Hi! 👋 

Thanks for submitting this PR! An issue I have is that because Slack notifications at present use attachments, when we add a text field the text is included twice: first as text and second as an attachment with a color.

It might be better if we set `text` to `tmpl(sn.Title)`?

**Comment by grobinson-grafana** [comment]
Hi! 👋

Thanks for submitting this PR to fix notifications in Slack! There is a small issue with this change in that the text is now included twice: first as text and second as an attachment with a color. This happens because Slack notifications at present use attachments, and when we add a text field both text and attachments are shown in the message. This is not the case when using the BlockKit API, but we have not updated Grafana to use BlockKit.

It might be better if we set text to tmpl(sn.Title)?

**Comment by grobinson-grafana** [comment]
Hi! 👋

Thanks for submitting this PR to fix notifications in Slack!

There is a small issue with this change in that the text is now included twice: first as text and second as an attachment with a color. This happens because Slack notifications at present use attachments, and when we add a text field both text and attachments are shown in the message.

This is not the case when using the BlockKit API, but we have not updated Grafana to use BlockKit.

It might instead be better if we set text to `tmpl(sn.Title)`?

**Comment by grobinson-grafana** [comment]
Hi! 👋

Thanks for submitting this PR to fix notifications in Slack!

There is a small issue with this change in that the text is now included twice: first as text and second as an attachment with a color. This happens because Slack notifications at present use attachments, and when we add a text field both text and attachments are shown in the message.

This is not the case when using the BlockKit API, but we have not updated Grafana to use BlockKit.

It might instead be better if we set text to `tmpl(sn.Title)`?

**Comment by grobinson-grafana** [comment]
Closing because a duplicate of https://github.com/grafana/grafana/pull/52391.

**Pull request closed without merging**
